### PR TITLE
Xero export settings

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/core/models/xero/xero-configuration/xero-export-settings.model.ts [hunk 1]
```diff
@@ -29,15 +29,173 @@ export type XeroExportSettingPost = {
...
+  }
+
+  static getExpenseGroupingOptions(): SelectFormOption[] {
+    return [
+      {
+        label: 'Report',
+        value: ExpenseGroupingFieldOption.CLAIM_NUMBER
+      },
+      {
+        label: 'Payment',
```
Comment (ashwin1111) [inline src/app/core/models/xero/xero-configuration/xero-export-settings.model.ts:67]: remove this option.

File: src/app/core/models/xero/xero-configuration/xero-export-settings.model.ts [hunk 2]
```diff
@@ -29,15 +29,173 @@ export type XeroExportSettingPost = {
...
+       label: 'Card Transaction Post date',
+       value: ExportDateType.POSTED_AT
+     }
+   ];
+ }
+
+  static getReimbursableExpenseStateOptions(isSimplifyReportClosureEnabled: boolean): SelectFormOption[] {
+    return [
+      {
+        label: isSimplifyReportClosureEnabled ? 'Processing' : 'Payment Processing',
```
Comment (ashwin1111) [inline src/app/core/models/xero/xero-configuration/xero-export-settings.model.ts:135]: remove isSimplifyReportClosureEnabled usage, let's default it to Processing. for 139 it's Closed

File: src/app/core/models/xero/xero-configuration/xero-export-settings.model.ts [hunk 3]
```diff
@@ -29,15 +29,173 @@ export type XeroExportSettingPost = {
...
+        label: isSimplifyReportClosureEnabled ? 'Closed' : 'Paid',
+        value: ExpenseState.PAID
+      }
+    ];
+  }
+
+  static getCCCExpenseStateOptions(isSimplifyReportClosureEnabled: boolean): SelectFormOption[] {
+    return [
+      {
+        label: isSimplifyReportClosureEnabled ? 'Approved' : 'Payment Processing',
```
Comment (ashwin1111) [inline src/app/core/models/xero/xero-configuration/xero-export-settings.model.ts:148]: same as above

File: src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts [hunk 4]
```diff
@@ -7,9 +20,130 @@ import { Component, OnInit } from '@angular/core';
...
+
+  windowReference: Window;
+
+  exportSettings:  XeroExportSettingGet;
+
+  employeeFieldMapping: EmployeeFieldMapping;
+
+  bankAccounts: DestinationAttribute[];
+
+  cccAccounts: DefaultDestinationAttribute[];
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts:39]: remove unused variables

File: src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts [hunk 5]
```diff
@@ -7,9 +20,130 @@ import { Component, OnInit } from '@angular/core';
...
+
+  accountsPayables: DefaultDestinationAttribute[];
+
+  vendors: DefaultDestinationAttribute[];
+
+  expenseAccounts: DefaultDestinationAttribute[];
+
+  isImportItemsEnabled: boolean;
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts:47]: not needed

File: src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts [hunk 6]
```diff
@@ -7,9 +20,130 @@ import { Component, OnInit } from '@angular/core';
...
+
+  expenseGroupByOptions =  XeroExportSettingModel.getExpenseGroupingOptions();
+
+  reimbursableExpenseGroupingDateOptions =  XeroExportSettingModel.getReimbursableExpenseGroupingDateOptions();
+
+  cccExpenseGroupingDateOptions = XeroExportSettingModel.getCCCExpenseGroupingDateOptions();
+
+  autoMapEmployeeTypes = XeroExportSettingModel.getAutoMapEmployeeOptions();
+
+  showNameInJournalOption: boolean;
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts:65]: not needed

File: src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts [hunk 7]
```diff
@@ -7,9 +20,130 @@ import { Component, OnInit } from '@angular/core';
...
+
+  EmployeeFieldMapping = EmployeeFieldMapping;
+
+  ConfigurationCtaText = ConfigurationCta;
+
+  readonly brandingFeatureConfig = brandingFeatureConfig;
+
+  readonly brandingContent = brandingContent.configuration.exportSetting;
+
+  is_simplify_report_closure_enabled: boolean = false;
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts:89]: not needed

File: src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts [hunk 8]
```diff
@@ -7,9 +20,130 @@ import { Component, OnInit } from '@angular/core';
...
+    this.isOnboarding = this.router.url.includes('onboarding');
+    const destinationAttributes = ['BANK_ACCOUNT'];
+
+    forkJoin([
+      this.exportSettingService.getExportSettings(),
+      this.mappingService.getGroupedDestinationAttributes(destinationAttributes, 'v1', 'xero')
+    ]).subscribe(response => {
+      this.exportSettings = response[0];
+      this.bankAccounts = response[1].BANK_ACCOUNT;
+      this.is_simplify_report_closure_enabled = (response[0].workspace_general_settings)?.is_simplify_report_closure_enabled;
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts:123]: not required

File: src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts [hunk 9]
```diff
@@ -7,9 +20,130 @@ import { Component, OnInit } from '@angular/core';
...
+    throw new Error('Method not implemented.');
+  }
+
+  constructPayloadAndSave($event: ConfigurationWarningOut) {
+    throw new Error('Method not implemented.');
+  }
+
+  setupPage() {
+    this.isOnboarding = this.router.url.includes('onboarding');
+    const destinationAttributes = ['BANK_ACCOUNT'];
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply src/app/integrations/xero/xero-shared/xero-export-settings/xero-export-settings.component.ts:115]: fixed in import settings PR

File: src/app/integrations/xero/xero-onboarding/xero-onboarding-routing.module.ts [hunk 10]
```diff
@@ -30,17 +30,17 @@ const routes: Routes = [
       {
         path: 'import_settings',
         component: XeroOnboardingImportSettingsComponent,
-        canActivate: [XeroTokenGuard, TenantGuard]
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/integrations/xero/xero-onboarding/xero-onboarding-routing.module.ts:33]: not fixed yet

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**